backend/de.metas.business/src/main/java-legacy/org/compiere/model/MPayment.java

```diff
@@ -1552,20 +1562,22 @@ public boolean allocateIt()
 	 *
 	 * @return true if allocated
 	 */
-	private boolean allocateInvoice()
+	private boolean allocateInvoice(@NonNull final InvoiceId invoiceId)
 	{
 		// 04627 begin (commented out old code)
 		// calculate actual allocation
 		// BigDecimal allocationAmt = getPayAmt(); // underpayment
 		// when allocating an invoice, we don't want to allocate more than the invoice's open amount
-		final I_C_Invoice invoice = getC_Invoice();
-		final BigDecimal invoiceOpenAmt = Services.get(IAllocationDAO.class).retrieveOpenAmt(invoice, false);
+		final I_C_Invoice invoice = InterfaceWrapperHelper.load(invoiceId, I_C_Invoice.class);
+
+		Check.errorIf(invoice == null, "Invoice cannot be null since C_Invoice_ID > 0, C_Invoice_ID = {}", invoiceId);
+		
+		final Money invoiceOpenAmt = Services.get(IAllocationDAO.class).retrieveOpenAmt(invoice, false);
```

Comment:
suggestion: refactor this line together with the lines above and introduce

IAllocationBL.retrieveOpenAmt(InvoiceId invoiceId, boolean creditMemoAdjusted)